modules/swagger-codegen/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -263,6 +263,7 @@ class ObjectSerializer
             return $deserialized;
         } else {
             // If a discriminator is defined and points to a valid subclass, use it.
+            $discriminator = $class::DISCRIMINATOR;
```

Comment:
This part is the important change. We get the discriminator out of the class to deserialise, instead of passing a potentially wrong discriminator through the whole deserialization recursion tree.